src/status_im/ui/screens/profile/events.cljs

```diff
@@ -39,43 +41,130 @@
+
+(handlers/register-handler-fx
+  :my-profile.drawer/update-name
+  (fn [{:keys [db]} [_ name]]
+    {:db (-> db
+             (assoc-in [:my-profile/drawer :valid-name?] (valid-name? name))
+             (assoc-in [:my-profile/drawer :name] name))}))
+
+(handlers/register-handler-fx
+  :my-profile.drawer/update-status
+  (fn [{:keys [db]} [_ status]]
+    (let [linebreak?  (string/includes? status "\n")
+          new-db      (if linebreak?
+                        (-> db
+                            (assoc-in [:my-profile/drawer :edit-status?] nil)
+                            (assoc-in [:my-profile/drawer :status] (utils/clean-text status)))
+                        (assoc-in db [:my-profile/drawer :status] status))]
+      (if linebreak?
+        {:db new-db
+         :dispatch [:my-profile.drawer/save-status]}
+        {:db new-db}))))
+
+(handlers/register-handler-fx
+  :my-profile/update-name
+  (fn [{:keys [db]} [_ name]]
+    {:db (-> db
+             (assoc-in [:my-profile/profile :valid-name?] (valid-name? name))
+             (assoc-in [:my-profile/profile :name] name))}))
+
+(handlers/register-handler-fx
+  :my-profile/update-status
+  (fn [{:keys [db]} [_ status]]
+    {:db (if (string/includes? status "\n")
+           (-> db
+               (assoc-in [:my-profile/profile :edit-status?] nil)
+               (assoc-in [:my-profile/profile :status] (utils/clean-text status)))
+           (assoc-in db [:my-profile/profile :status] status))}))
+
 (handlers/register-handler-fx
   :my-profile/update-picture
   (fn [{:keys [db]} [this-event base64-image]]
     (if base64-image
-      {:db (assoc-in db [:my-profile/edit :photo-path] (str "data:image/jpeg;base64," base64-image))}
+      {:db (assoc-in db [:my-profile/profile :photo-path] (str "data:image/jpeg;base64," base64-image))}
       {:open-image-picker this-event})))
 
+(defn clean-name [{:accounts/keys [current-account-id] :as db} edit-view]
+  (let [name (get-in db [edit-view :name])]
+    (if (valid-name? name)
+      name
+      (get-in db [:accounts/accounts current-account-id :name]))))
+
+(handlers/register-handler-fx
+  :my-profile.drawer/save-name
+  (fn [{:keys [db]} _]
+    (let [cleaned-name (clean-name db :my-profile/drawer)]
+      {:db (-> db
+               (assoc-in [:my-profile/drawer :name] cleaned-name)
+               (assoc-in [:my-profile/drawer :valid-name?] true)
+               (dissoc :my-profile/profile))
+       :dispatch [:account-update {:name cleaned-name}]})))
+
+(handlers/register-handler-fx
+  :my-profile.drawer/save-status
+  (fn [{:keys [db]} _]
+    (let [status (get-in db [:my-profile/drawer :status])
+          new-db (-> db
+                     (assoc-in [:my-profile/drawer :edit-status?] nil)
+                     (dissoc :my-profile/profile))]
+      (if (string/blank? status)
+        {:db new-db}
+        {:db new-db
+         :dispatch-n [[:check-status-change status]
+                      [:account-update {:status status}]]}))))
+
```

Comment:
Ideally we would like to have `:check-status-change` and `:account-update` refactored into pure functions and just call them there instead of the dispatch, but I understand it's not focus of this PR and it's another module (`accounts`), so it can be done later.

Reply:
yes it's an old part of the code where there is still ui/side-effects